presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/BingTileFunctions.java

```diff
@@ -85,6 +88,29 @@
 
     private BingTileFunctions() {}
 
+    @Description("Encodes a Bing tile into a bigint")
+    @ScalarOperator(CAST)
+    @SqlType(StandardTypes.BIGINT)
+    public static long castToBigint(@SqlType(BingTileType.NAME) long tile)
+    {
+        return tile;
+    }
+
+    @Description("Decodes a Bing tile from a bigint")
+    @ScalarOperator(CAST)
+    @SqlType(BingTileType.NAME)
+    public static long castFromBigint(@SqlType(StandardTypes.BIGINT) long tile)
+    {
+        try {
+            BingTile.decode(tile);
```

Comment:
follow up: Currently `BingTile.decode` must create an object of `BingTile`. The main reason for calling `BingTile.decode` is to validate the tile stored in a long. For efficiency I would recommend to have a dedicated method, e.g.: `BingTile.validate` or smthng.